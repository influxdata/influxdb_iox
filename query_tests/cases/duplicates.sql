-- Test for predicate push down explains
-- IOX_SETUP: OneMeasurementThreeChunksWithDuplicates

-- Plan with order by
explain verbose select time, state, city, min_temp, max_temp, area from h2o order by time, state, city;


-- plan without order by
explain verbose select time, state, city, min_temp, max_temp, area from h2o;

-- Union plan
EXPLAIN VERBOSE select state as name from h2o UNION ALL select city as name from h2o;
