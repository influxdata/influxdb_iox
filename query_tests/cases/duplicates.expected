-- Test Setup: OneMeasurementThreeChunksWithDuplicates
-- SQL: explain verbose select time, state, city, min_temp, max_temp, area from h2o order by time, state, city;
+-----------------------------------------+-------------------------------------------------------------------------------------------------+
| plan_type                               | plan                                                                                            |
+-----------------------------------------+-------------------------------------------------------------------------------------------------+
| logical_plan                            | Sort: #time ASC NULLS FIRST, #state ASC NULLS FIRST, #city ASC NULLS FIRST                      |
|                                         |   Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |     TableScan: h2o projection=None                                                              |
| logical_plan after projection_push_down | Sort: #time ASC NULLS FIRST, #state ASC NULLS FIRST, #city ASC NULLS FIRST                      |
|                                         |   Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| logical_plan after simplify_expressions | Sort: #time ASC NULLS FIRST, #state ASC NULLS FIRST, #city ASC NULLS FIRST                      |
|                                         |   Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| logical_plan after projection_push_down | Sort: #time ASC NULLS FIRST, #state ASC NULLS FIRST, #city ASC NULLS FIRST                      |
|                                         |   Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| logical_plan after simplify_expressions | Sort: #time ASC NULLS FIRST, #state ASC NULLS FIRST, #city ASC NULLS FIRST                      |
|                                         |   Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |     TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| physical_plan                           | SortExec: [time ASC,state ASC,city ASC]                                                         |
|                                         |   ProjectionExec: expr=[time, state, city, min_temp, max_temp, area]                            |
|                                         |     ExecutionPlan(PlaceHolder)                                                                  |
|                                         |       DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |         SortPreservingMergeExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST] |
|                                         |           ExecutionPlan(PlaceHolder)                                                            |
|                                         |             SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |               IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |             SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |               IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |       DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |         SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |           IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
|                                         |       DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |         SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |           IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
+-----------------------------------------+-------------------------------------------------------------------------------------------------+
-- SQL: explain verbose select time, state, city, min_temp, max_temp, area from h2o;
+-----------------------------------------+-----------------------------------------------------------------------------------------------+
| plan_type                               | plan                                                                                          |
+-----------------------------------------+-----------------------------------------------------------------------------------------------+
| logical_plan                            | Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |   TableScan: h2o projection=None                                                              |
| logical_plan after projection_push_down | Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| logical_plan after simplify_expressions | Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| logical_plan after projection_push_down | Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| logical_plan after simplify_expressions | Projection: #time, #state, #city, #min_temp, #max_temp, #area                                 |
|                                         |   TableScan: h2o projection=Some([0, 1, 2, 3, 4, 5])                                          |
| physical_plan                           | ProjectionExec: expr=[time, state, city, min_temp, max_temp, area]                            |
|                                         |   ExecutionPlan(PlaceHolder)                                                                  |
|                                         |     DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |       SortPreservingMergeExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST] |
|                                         |         ExecutionPlan(PlaceHolder)                                                            |
|                                         |           SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |           SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |     DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |       SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |         IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
|                                         |     DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |       SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |         IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
+-----------------------------------------+-----------------------------------------------------------------------------------------------+
-- SQL: EXPLAIN VERBOSE select state as name from h2o UNION ALL select city as name from h2o;
+-----------------------------------------+---------------------------------------------------------------------------------------------------+
| plan_type                               | plan                                                                                              |
+-----------------------------------------+---------------------------------------------------------------------------------------------------+
| logical_plan                            | Union                                                                                             |
|                                         |   Projection: #state AS name                                                                      |
|                                         |     TableScan: h2o projection=None                                                                |
|                                         |   Projection: #city AS name                                                                       |
|                                         |     TableScan: h2o projection=None                                                                |
| logical_plan after projection_push_down | Union                                                                                             |
|                                         |   Projection: #state AS name                                                                      |
|                                         |     TableScan: h2o projection=Some([4])                                                           |
|                                         |   Projection: #city AS name                                                                       |
|                                         |     TableScan: h2o projection=Some([1])                                                           |
| logical_plan after simplify_expressions | Union                                                                                             |
|                                         |   Projection: #state AS name                                                                      |
|                                         |     TableScan: h2o projection=Some([4])                                                           |
|                                         |   Projection: #city AS name                                                                       |
|                                         |     TableScan: h2o projection=Some([1])                                                           |
| logical_plan after projection_push_down | Union                                                                                             |
|                                         |   Projection: #state AS name                                                                      |
|                                         |     TableScan: h2o projection=Some([4])                                                           |
|                                         |   Projection: #city AS name                                                                       |
|                                         |     TableScan: h2o projection=Some([1])                                                           |
| logical_plan after simplify_expressions | Union                                                                                             |
|                                         |   Projection: #state AS name                                                                      |
|                                         |     TableScan: h2o projection=Some([4])                                                           |
|                                         |   Projection: #city AS name                                                                       |
|                                         |     TableScan: h2o projection=Some([1])                                                           |
| physical_plan                           | ExecutionPlan(PlaceHolder)                                                                        |
|                                         |   ProjectionExec: expr=[state as name]                                                            |
|                                         |     ExecutionPlan(PlaceHolder)                                                                    |
|                                         |       ProjectionExec: expr=[state]                                                                |
|                                         |         DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |           SortPreservingMergeExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST] |
|                                         |             ExecutionPlan(PlaceHolder)                                                            |
|                                         |               SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |               SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |       ProjectionExec: expr=[state]                                                                |
|                                         |         DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |           SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
|                                         |       ProjectionExec: expr=[state]                                                                |
|                                         |         DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |           SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
|                                         |   ProjectionExec: expr=[city as name]                                                             |
|                                         |     ExecutionPlan(PlaceHolder)                                                                    |
|                                         |       ProjectionExec: expr=[city]                                                                 |
|                                         |         DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |           SortPreservingMergeExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST] |
|                                         |             ExecutionPlan(PlaceHolder)                                                            |
|                                         |               SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |               SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]            |
|                                         |                 IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                   |
|                                         |       ProjectionExec: expr=[city]                                                                 |
|                                         |         DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |           SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
|                                         |       ProjectionExec: expr=[city]                                                                 |
|                                         |         DeduplicateExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]           |
|                                         |           SortExec: [city ASC NULLS LAST,state ASC NULLS LAST,time ASC NULLS LAST]                |
|                                         |             IOxReadFilterNode: table_name=h2o, chunks=1 predicate=Predicate                       |
+-----------------------------------------+---------------------------------------------------------------------------------------------------+
